contracts/test/BLSSig.t.sol

```diff
@@ -0,0 +1,82 @@
+// SPDX-License-Identifier: Unlicensed
+
+/* solhint-disable contract-name-camelcase, func-name-mixedcase, one-contract-per-file */
+
+pragma solidity ^0.8.0;
+
+// Libraries
+import "forge-std/Test.sol";
+import { BN254 } from "bn254/BN254.sol";
+
+// Target contract
+import { BLSSig } from "../src/libraries/BLSSig.sol";
+
+contract BLSSig_Test is Test {
+    function genBLSSig(bytes memory message)
+        private
+        returns (BN254.G2Point memory, BN254.G1Point memory)
+    {
+        // Generate a BLS verification key and signature  using rust code
+        string[] memory cmds = new string[](3);
+        cmds[0] = "diff-test";
+        cmds[1] = "gen-bls-sig";
+        cmds[2] = vm.toString(message);
+
+        bytes memory result = vm.ffi(cmds);
+
+        (BN254.G2Point memory vk, BN254.G1Point memory sig) =
+            abi.decode(result, (BN254.G2Point, BN254.G1Point));
+        return (vk, sig);
+    }
+
+    function testFuzz_BLS_hashes_computation(bytes memory input) external {
+        //bytes memory input = "Message";
+
+        string[] memory cmds = new string[](3);
+        cmds[0] = "diff-test";
+        cmds[1] = "gen-bls-hashes";
+        cmds[2] = vm.toString(input);
+
+        bytes memory result = vm.ffi(cmds);
+
+        (uint256 fieldElem, BN254.G1Point memory hashToCurveElem) =
+            abi.decode(result, (uint256, BN254.G1Point));
+
+        assertEq(fieldElem, BLSSig.hashToField(input));
+        assertEq(abi.encode(hashToCurveElem), abi.encode(BLSSig.hashToCurve(input)));
+    }
+
+    function test_SigVerification_Succeeds() external {
+        bytes memory message = "Hi";
+        (BN254.G2Point memory vk, BN254.G1Point memory sig) = genBLSSig(message);
+        BLSSig.verifyBlsSig(message, sig, vk);
+    }
+
+    // This is due to a current limitation/bug of foundry. See
+    // https://github.com/foundry-rs/foundry/issues/4405
+    function wrapVerifyBlsSig(
```

Comment:
very good, let's create an issue link that upstream issue so we will be reminded to address it later.

Reply:
Yes, ticket created: https://github.com/EspressoSystems/espresso-sequencer/issues/847.